addons/web_editor/static/src/js/common/column_layout_mixin.js

```diff
@@ -0,0 +1,144 @@
+/** @odoo-module **/
+"use strict";
+
+export const ColumnLayoutMixin = {
+    /**
+     * Calculates the number of columns for the mobile or desktop version.
+     * If all elements don't have the same size, returns null.
+     *
+     * @private
+     * @param {HTMLCollection} columnEls - elements in the .row container
+     * @param {boolean} isMobile
+     * @returns {integer|null} number of columns
+     */
+    _getNbColumns(columnEls, isMobile) {
+        if (!columnEls) {
+            return 0;
+        }
+
+        const resolutionModifier = isMobile ? "" : "lg-";
+        const screenType = isMobile ? "mobile" : "desktop";
+        const colRegex = new RegExp(`(?<!\\S)col-${resolutionModifier}(\\d{1,2})(?!\\S)`);
+        const colSize = parseInt(columnEls[0].className.match(colRegex)?.[1] || 12);
+
+        const isCustomized =
+            columnEls[0].parentElement.classList.contains(`o_we_${screenType}-size-custom`);
```

Comment:
- Don't mix _ and - in class names
- Don't build class name when you are not forced to: `isMobile ? o_we_mobile_custom_size : ...`. Eases grep and reduces mistakes.
- This class should be documented somewhere. Either here or probably rather near the code that adds it. I actually remember we spoke about it but I can't remember what was decided or why it is needed: a comment in the code should :+1:

Reply:
These classes are needed because controlling whether all items have the same size is not enough with "weirder" cases, such as when the offsets have been meddled with (e.g. all items are `col-4`, but the 1st item has an `offset-2` set by the user: it should show "custom", not "3" columns). Likewise, I can't simply control the offsets to know if they've been altered, because there are default offsets, for instance with 5 columns (`offset-1` on the 1st, `offset-5` on the 6th), or with 3 columns and 5 items, and no easy way to differentiate them from custom ones.

But actually, this also means that the `allColSizesEqual` variable just after this seems useless after all, because if they're not equal, they must be customized.